### PR TITLE
[10.x] Add float validation rule

### PR DESCRIPTION
**Pull request by fuwasegu**
## Introduction
While Laravel offers validation rules for various types like integer and bool, there currently is no native validation rule for float. Given that the existing numeric rule allows for more lenient conditions such as whitespaces in numeric strings (" 123 ") and scientific notation (e.g., "123e0"), a more strict validation specifically for floats is needed.

## Implementation
~~In an attempt to align this new float validation with the existing integer rule, the implementation also considers strings that can be evaluated as floats to return true.~~

~~The function utilizes PHP's filter_var with the FILTER_VALIDATE_FLOAT flag. However, filter_var($value, FILTER_VALIDATE_FLOAT) alone would allow integers to pass through, which is not what we aim for in a float validation. Therefore, an additional check for !is_int($value) has been added to ensure only floats (or float-like strings) are considered valid.~~

Initially, the idea was to restrict the validation to only allow true floating-point numbers, excluding integers. However, to align better with PHP's native filter_var behavior, we decided to keep it simple and straightforward.

```php
public function validateFloat($attribute, $value)
{
    return filter_var($value, FILTER_VALIDATE_FLOAT) !== false;
}
```

This implementation uses PHP's filter_var with the FILTER_VALIDATE_FLOAT flag. This follows PHP's built-in logic and will include integers as valid floats, aligning it with PHP's general type loose behavior.

### CODE REVIEW

**Comment by henzeb** [comment]
What about https://laravel.com/docs/10.x/validation#rule-decimal?

**Comment by fuwasegu** [comment]
Thank you for pointing out the existing decimal:min,max rule. While this rule does offer the ability to validate floating-point numbers, it also introduces a couple of challenges:

Consistency: Laravel provides native rules for integer, boolean, etc., making a float-specific rule a logical addition for consistency.

Clarity in Error Messages: Having a float-specific validation rule would enable us to provide more precise and helpful error messages tailored for floating-point numbers.

Intuitiveness: Using decimal:min,max for floats might not be as intuitive for some developers who expect to see a dedicated rule for floating-point numbers.

In summary, while decimal:min,max can serve a similar purpose, a dedicated float validation rule would enhance usability, clarity, and consistency within Laravel's validation features.

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.
